tiledb/sm/query/readers/dense_reader.h

```diff
@@ -265,6 +271,136 @@ class DenseReader : public ReaderBase, public IQueryStrategy {
 
   /** Perform necessary checks before exiting a read loop */
   Status complete_read_loop();
```

Comment:
Note to reviewer: these were copied from reader_base.h but modified to pass in the query condition results and also remove the return values as overflows are detected earlier and returned through the read state.